python/astro_metadata_translator/file_helpers.py

```diff
@@ -0,0 +1,248 @@
+        """Read a FITS header using astropy.
+
+        Parameters
+        ----------
+        file : `str`
+            The file to read.
+        hdu : `int`
+            The header number to read.
+        can_raise : `bool`, optional
+            Indicate whether the function can raise and exception (default)
+            or should return `None` on error. Can still raise if an unexpected
+            error is encountered.
+
+        Returns
+        -------
+        md : `dict`
+            The requested header. `None` if it could not be read and
+            ``can_raise`` is `False`.
+        """
+        header = None
+        try:
+            with fits.open(file) as fits_file:
+                try:
+                    header = fits_file[hdu].header
+                except IndexError as e:
+                    if can_raise:
+                        raise e
+        except Exception as e:
+            if can_raise:
+                raise e
+        return header
+
+
+def find_files(files, regex):
+    """Find files for processing.
+
+    Parameters
+    ----------
+    files : iterable of `str`
+        The files or directories from which the headers are to be read.
+    regex : `str`
+        Regular expression string used to filter files when a directory is
+        scanned.
+    """
+    file_regex = re.compile(regex)
+    found_files = []
+
+    # Find all the files of interest
+    for file in files:
+        if os.path.isdir(file):
+            for root, dirs, files in os.walk(file):
+                for name in files:
+                    path = os.path.join(root, name)
+                    if os.path.isfile(path) and file_regex.search(name):
+                        found_files.append(path)
+        else:
+            found_files.append(file)
+
+    return found_files
+
+
+def read_basic_metadata_from_file(file, hdrnum, errstream=sys.stderr, can_raise=False):
+    """Read a raw header from a file, merging if necessary
+
+    Parameters
+    ----------
+    file : `str`
+        Name of file to read. Can be FITS or YAML. YAML must be a simple
+        top-level dict.
+    hdrnum : `int`
+        Header number to read. Only relevant for FITS. If greater than 1
+        it will be merged with the primary header.
+    errstream : `io.StringIO`, optional
+        Stream to send messages that would normally be sent to standard
+        error. Defaults to `sys.stderr`. Only used if exceptions are disabled.
+    can_raise : `bool`, optional
+        Indicate whether the function can raise and exception (default)
+        or should return `None` on error. Can still raise if an unexpected
+        error is encountered.
+
+    Returns
+    -------
+    header : `dict`
+        The header as a dict. Can be `None` if there was a problem reading
+        the file.
+    """
+    if file.endswith(".yaml"):
+        md = read_test_file(file,)
+        if hdrnum != 0:
+            # YAML can't have HDUs
+            hdrnum = 0
```

Comment:
Nothing else is done to `md` so we should just return the value here

```suggestion
        return md
```

Reply:
It is used later to skip the merging step.